apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -32,9 +33,40 @@ defmodule Aecore.Structures.SignedTx do
     key == nil && signature == nil
   end
 
+  @spec valid_byte_size?(DataTx.t()) :: boolean()
+  def valid_byte_size?(%DataTx{payload: %{oracle_address: address}}) do
+    if Wallet.key_size_valid?(address) do
+      true
+    else
+      Logger.error("Wrong oracle address size")
+      false
+    end
+  end
+
+  @spec valid_byte_size?(DataTx.t()) :: boolean()
+  def valid_byte_size?(%DataTx{
```

Comment:
this check should be in data tx file

Reply:
Maybe in SpendTx? Because in this function I check the byte_size for SpendTx, and DataTx is something common for all txs types.

Reply:
Another way, I can put byte_size validations for all types of txs in DataTx module. Since it is common for all txs types, probably it would be better to handle all the checks there.

Reply:
each check should be in the module where the field is defined initially